### PR TITLE
Easier loop length quantization

### PR DESCRIPTION
**Pull request by 21echoes**
I often find that the loop length gets "off" from a nice even quantization. E.g., "15.47 beats". When in this state, there's no way to get it back to a round number if your loop length resolution is set to something in beats (1 beat, 1 bar, etc.). Scrolling left gets you down to e.g. "0.47 beats", scrolling up gets you to e.g. "20.47 beats", and you never end up at a nice "16 beats".

The proposed fix: when scrolling down, if the loop length resolution is something in beats, let it go down to exactly 1 "resolution value". So e.g. if the user set their resolution to 1 beat, they'd scroll from 15.47 beats down to 14.57, 13.57, ..., 1.47, then to 1.00 beats. From there they can scroll back up in quantized loop lengths: 1 beat, 2 beats, etc.

### CODE REVIEW

File: lib/encoder_actions.lua [hunk 1]
```diff
@@ -1186,15 +1186,23 @@ function ea.move_rec_end(d)
   else
     res = d/rec_loop_enc_resolution
   end
-  if d <= 0 and util.round(rec[rec.focus].start_point,0.01) < util.round(rec[rec.focus].end_point + ((res)/lbr[params:get("live_buff_rate")]),0.01) then
-    rec[rec.focus].end_point = util.clamp(rec[rec.focus].end_point+((res)/lbr[params:get("live_buff_rate")]),(1+(8*(rec.focus-1))),(9+(8*(rec.focus-1))))
+  local min_end_point = 1+(8*(rec.focus-1))
+  local max_end_point = 9+(8*(rec.focus-1))
+  local resolved_res = res / lbr[params:get("live_buff_rate")]
+  local new_end_point = rec[rec.focus].end_point + resolved_res
```
Comment (21echoes) [inline lib/encoder_actions.lua:1192]: i did some mild cleanup here as well, extracting these re-used statements like `1+(8*(rec.focus-1))` into proper variables for clearer & easier re-use through the next few lines.

File: lib/encoder_actions.lua [hunk 2]
```diff
@@ -1186,15 +1186,23 @@ function ea.move_rec_end(d)
...
+  if params:get("rec_loop_enc_resolution") >= 3 then
+    -- When quantizing the loop endpoint, and the user tries to make it very small
+    -- let it only go to one quantization value at minimum
+    -- (instead of the very beginning of the loop)
+    min_end_point = rec[rec.focus].start_point + 1 / (rec_loop_enc_resolution * lbr[params:get("live_buff_rate")])
+  end
```
Comment (21echoes) [inline lib/encoder_actions.lua:1198]: these 3 lines of code + 3 lines of comments are the only "actual" changes

File: lib/encoder_actions.lua (hunk 1 above)
Comment (dndrks) [reply lib/encoder_actions.lua:1192]: raaaad, tytyty!!

File: lib/encoder_actions.lua (hunk 2 above)
Comment (dndrks) [reply lib/encoder_actions.lua:1198]: absolutely wonderful, thank you for this very thoughtful contribution!!!

**Review by dndrks — APPROVED** [review]
wowowow, @21echoes , thank you so much!! idk how i missed this PR, but thank you for solving this issue <33

**Pull request merged**